### PR TITLE
SQL/JDBC implementation

### PR DESCRIPTION
**Pull request by srfarley**
@joeferner: I intend to squash the commits before merge.

### CODE REVIEW

File: sql/src/main/java/org/vertexium/sql/SqlGraphConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        dataSource.setUrl(url);
+        dataSource.setUsername(username);
+        dataSource.setPassword(password);
+        Handle handle = new DBI(dataSource).open();
+        vertexMap = newVertexMap(handle);
+        edgeMap = newEdgeMap(handle);
+        metadataMap = newMetadataMap(handle);
+    }
+
+    private VertexiumSerializer createSerializer(Map<String, Object> config) {
```
Comment (joeferner) [inline sql/src/main/java/org/vertexium/sql/SqlGraphConfiguration.java]: I know I put this here but should we move this to GraphConfiguration similar to the table name moving there?

File: sql/src/main/java/org/vertexium/sql/SqlGraphConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        String driverClass = getConfigString(config, CONFIG_JDBC_DRIVER_CLASS);
+        String url = getConfigString(config, CONFIG_JDBC_URL);
+        String username = getConfigString(config, CONFIG_JDBC_USERNAME);
+        String password = getConfigString(config, CONFIG_JDBC_PASSWORD);
+        dataSource = new BasicDataSource();
+        dataSource.setDriverClassName(driverClass);
+        dataSource.setUrl(url);
+        dataSource.setUsername(username);
+        dataSource.setPassword(password);
+        Handle handle = new DBI(dataSource).open();
```
Comment (joeferner) [inline sql/src/main/java/org/vertexium/sql/SqlGraphConfiguration.java]: I find it a little strange to open the Handle here? Should this be moved to SqlGraph?

**Comment by joeferner** [comment]
Can you also update the CHANGELOG.md.  Other than that it looks good.

File: sql/src/main/java/org/vertexium/sql/SqlGraphConfiguration.java (hunk 2 above)
Comment (srfarley) [inline sql/src/main/java/org/vertexium/sql/SqlGraphConfiguration.java]: I didn't want to have a dependency on the JDBI library anywhere else but here. But having reviewed this more, I think I need to create a new `Handle` instance within every call in `SqlMap` (wrapped by an auto-closing try block). That means I'll need to provide `SqlMap` with a `DBI` instance.

File: sql/src/main/java/org/vertexium/sql/SqlGraphConfiguration.java (hunk 1 above)
Comment (srfarley) [inline sql/src/main/java/org/vertexium/sql/SqlGraphConfiguration.java]: done

File: sql/src/main/java/org/vertexium/sql/SqlGraphConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+    protected SqlMap<Object> metadataMap() {
+        return metadataMap;
+    }
+
+    protected String tableNameWithPrefix(String tableName) {
+        return getTableNamePrefix() + "_" + tableName;
+    }
+
+    private SqlMap<InMemoryTableElement<InMemoryEdge>> newEdgeMap() {
```
Comment (joeferner) [inline sql/src/main/java/org/vertexium/sql/SqlGraphConfiguration.java]: This still seems like an odd place for these methods. Shouldn't these be in SqlGraph?

**Pull request closed without merging**